File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Utils/Determinator.cs
using KingstonDynamics.DragonSong.Protocol.Messaging.Data;

namespace KingstonDynamics.DragonSong.Protocol.Utils
{
    public static class Determinator
    {
        public static MessageType Determinate(byte[] data)
        {
            // Get Reader
            var reader = new Readinator(data);
            
            // Determine Value 
            return (MessageType) reader.ReadInt();
        }
    }
}

File: src/old/DragonSong-Protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/utils/Determinator.java
package com.kingstondynamics.dragonsong.protocol.utils;

import com.kingstondynamics.dragonsong.protocol.message.data.MessageType;

public class Determinator {
    
    /**
     * Private Constructor for Utlity Class
     */
    private Determinator() {
        throw new AssertionError();
    }
    
    public static MessageType determinate(byte[] data) {
        
        // Load Data to be Read
        Readinator reader = new Readinator(data);
        
        // Check first four bytes
        int opcode = reader.readInt();
        
        // Return Type
        return MessageType.fromValue(opcode);
    }
}
